src/clj/puppetlabs/services/jruby/jruby_core.clj

```diff
@@ -142,26 +142,37 @@
 ;;; Public
 
 (schema/defn ^:always-validate
+  initialize-lifecycle-fns :- jruby-schemas/LifecycleFns
+  [config :- (schema/maybe {schema/Keyword schema/Any})]
+  (-> config
+      (update-in [:initialize] #(or % identity))
+      (update-in [:shutdown] #(or % identity))
+      (update-in [:shutdown-on-error] #(or % (fn [f] (f))))
```

Comment:
I wasn't entirely sure whether we actually want to have defaults for these three or whether they should always have to be provided, but I went with provide defaults for now, and curious whether other folks have opinions.